backends/platform/ios7/ios7_osys_video.mm

```diff
@@ -160,6 +164,56 @@ static inline void execute_on_main_thread(void (^block)(void)) {
 	}
 }
 
+#if TARGET_OS_IOS
+void OSystem_iOS7::setSupportedScreenOrientation(ScreenOrientation screenOrientation) {
+	switch (screenOrientation) {
+	case kScreenOrientationPortrait:
+	case kScreenOrientationFlippedPortrait:
+		[[iOS7AppDelegate iPhoneView] setSupportedScreenOrientations:UIInterfaceOrientationMaskPortrait];
+		break;
+	case kScreenOrientationLandscape:
+	case kScreenOrientationFlippedLandscape:
+		[[iOS7AppDelegate iPhoneView] setSupportedScreenOrientations:UIInterfaceOrientationMaskLandscape];
+		break;
+	case kScreenOrientationAuto:
+	default:
+		[[iOS7AppDelegate iPhoneView] setSupportedScreenOrientations:UIInterfaceOrientationMaskAll];
+		break;
+	}
+
+	// Trigger the new orientation
+#ifdef __IPHONE_16_0
+	if (@available(iOS 16.0, *)) {
+		execute_on_main_thread(^ {
+			[UIViewParentController([iOS7AppDelegate iPhoneView]) setNeedsUpdateOfSupportedInterfaceOrientations];
+		});
+	} else
+#endif
+	{
+		switch (screenOrientation) {
+		case kScreenOrientationPortrait:
+		case kScreenOrientationFlippedPortrait:
+			execute_on_main_thread(^ {
+				[[UIDevice currentDevice] setValue:@(UIInterfaceOrientationPortrait) forKey:@"orientation"];
+			});
+			break;
+		case kScreenOrientationLandscape:
+		case kScreenOrientationFlippedLandscape:
+			execute_on_main_thread(^ {
+				[[UIDevice currentDevice] setValue:@(UIInterfaceOrientationLandscapeRight) forKey:@"orientation"];
+			});
+			break;
+		case kScreenOrientationAuto:
+		default:
+			break;
+		}
+	}
+	// Update the screen orientation value to make sure the keyboard isn't toggled when
+	// starting a game in landscape
+	_screenOrientation = screenOrientation;
+}
```

Comment:
For this block above (for iOS < 16) shouldn't you check that the current orientation is valid for the new mask and do nothing if that is the case?
It seems to me that for example if the current orientation is `UIInterfaceOrientationLandscapeLeft` and the new mask is `UIInterfaceOrientationMaskLandscape` it will rotate to `UIInterfaceOrientationLandscapeRight` when you would probably not expect any rotation.

Reply:
Also you probably do not want to set `_screenOrientation` if `screenOrientation` is `kScreenOrientationAuto`?

And while the function is only called with `kScreenOrientationPortrait`, `kScreenOrientationLandscape` and `kScreenOrientationAuto`, since the implementation above also includes the cases for `kScreenOrientationFlippedPortrait` and `kScreenOrientationFlippedLandscape`, you may want to be consistent for those two cases as well between the "orientation" value set in the switch above, and the value set in `_screenOrientation`.

Reply:
Good comments.
Updated according to comments.
Regarding the last comment, UIInterfaceOrientation is a UIKit enum which we don't want to include in the event handler. So that's why you probably handle them as pure integers. We could probably change the type to send ScreenOrientation enum values instead. That would be an easy refactor that we can do whenever.

Reply:
> Regarding the last comment, UIInterfaceOrientation is a UIKit enum which we don't want to include in the event handler. So that's why you probably handle them as pure integers.

Ha yes. That code is c++, so we cannot include objective-C headers there.
I think just a comment would be good enough actually.
```c++
case 3: // UIDeviceOrientationLandscapeLeft
	newOrientation = kScreenOrientationLandscape;
	break;
```
Or we could indeed do the mapping in `ios7_video.mm` and pass a `ScreenOrientation` value for the `kInputOrientationChanged` event.

And it seems that you are inconsistent with what we do in `handleEvent_orientationChanged` in your code above. If I read the enum correctly `UIInterfaceOrientationLandscapeRight` is `4` and is mapped to `kScreenOrientationFlippedLandscape` and not `kScreenOrientationLandscape` as you do here (or maybe you should set "orientation" to `UIDeviceOrientationLandscapeLeft` and continue using `kScreenOrientationLandscape`?).